src/shipit_signoff/shipit_signoff/api.py

```diff
@@ -183,40 +180,85 @@ def sign_off(uid):
+    if step.policy_data['method'] != 'local':
+        logger.error('Step %s has unknown policy method: %s',
+                     uid, step.policy_data['method'])
+        abort(400, 'Policy method not known')
+
     try:
-        check_whether_policy_can_be_signed(email, claim_group, policy_definition, existing_signatures)
+        check_whether_policy_can_be_signed(
+            email, claim_group, policy_definition, existing_signatures)
     except UnauthorizedUserError as e:
+        logger.error('User %s (%s) not allowed to sign step %s',
+                     email, claim_group, uid)
         abort(403, str(e))
     except NoSignoffLeftError as e:
+        logger.error(
+            'Step %s already fully signed-off (user %s attempting)', uid, email)
         abort(409, str(e))
 
     insert_new_signature(step, email, claim_group)
     all_signatures = step.signatures
 
     if is_sign_off_policy_met(policy_definition, all_signatures):
-        # TODO: Do something more useful
-        print('Step fully signed off!')
+        step.state == SigningStatus.completed
+        db.session.commit()
+        logger.info('Step %s fully signed off!', uid)
 
     return {}
 
 
+@moz_accept_token()
 def delete_signature(uid):
     '''
-    TODO: implement
+    Delete a signature from a signoff step.
     '''
     logger.info('Removing signature from step %s', uid)
-    if uid not in STEPS:
-        return None, 404
-    SIGNOFFS[uid] = False
-    STEPS[uid] = 'running'
-    return None
+    try:
+        step = get_step_by_uid(uid)
+    except NoResultFound:
+        logger.error('No such step found %s', uid)
+        abort(404)
+    email = get_logged_in_email()
+    claim_group = request.json['group']
+    if not is_user_in_group(claim_group):
+        logger.error(
+            'User %s is not in the group %s when deleting step %s', email, claim_group, uid)
+        abort(403)
+    existing_signatures = step.signatures
+    if not existing_signatures:
+        logger.error(
+            'No signatures on step %s when trying to remove %s', uid, email)
+        abort(409)
+    if step.policy_data['method'] != 'local':
+        logger.error('Unknown policy method for step %s: %s',
+                     uid, step.policy_data['method'])
+        abort(400, 'Unknown policy method')
+    policy_definition = step.policy_data['definition']
+    try:
+        check_whether_policy_can_be_unsigned(
+            email, claim_group, policy_definition, existing_signatures)
+    except UnauthorizedUserError as e:
+        logger.error(
+            'User %s not permitted to remove signature from step %s', email, uid)
+        abort(403, str(e))
+    except NoSignaturePresentError as e:
+        logger.error(
+            'User %s attempting to remove missing signature from step %s', email, uid)
+        abort(409, str(e))
+    delete_existing_signature(step, email, claim_group)
+    if not is_sign_off_policy_met(policy_definition, step.signatures):
```

Comment:
This block seems to imply that a signature can be removed from a complete step. Is that true? It doesn't seem safe to do that, as downstream steps (or anything watching) may have already taken action based on the step being completed.